security-insights-schema-1.0.0.yaml

```diff
@@ -251,8 +251,82 @@ properties:
                 required:
                 - threat-model-created
                 type: object
-        required:
-        - threat-model
+            self-assessment:
+                $id: '#/properties/security-artifacts/properties/self-assessment'
+                additionalProperties: false
+                properties:
+                    self-assessment-created:
+                        $id: '#/properties/security-artifacts/properties/self-assessment/properties/self-assessment-created'
+                        description: 'Define whether a security self assessment for the project has been created. A false value may be used to add comments regarding the status of the assessment.'
+                        type: boolean
+                    evidence-url:
+                        $id: '#/properties/security-artifacts/properties/self-assessment/properties/evidence-url'
+                        additionalItems: false
+                        items:
+                            $id: '#/properties/security-artifacts/properties/self-assessment/properties/evidence-url/items'
+                            anyOf:
+                            -   $id: '#/properties/security-artifacts/properties/self-assessment/properties/evidence-url/items/anyOf/0'
+                                description: 'Link to the evidence of the security self assessment.'
+                                type: string
+                                format: iri
+                                pattern: '^https?:\/\/'
+                        type: array
+                        uniqueItems: true
+                    comment:
+                        $id: '#/properties/security-artifacts/properties/self-assessment/properties/comment'
+                        description: 'Additional context regarding the security self assessment or its status. Maximum length 560 chars.'
+                        type: string
+                        pattern: '^(.|\n){1,560}$'
+                if:
+                    properties:
+                        self-assessment-created:
+                            const: true
+                then:
+                    required:
+                    - evidence-url
+                required:
+                - self-assessment-created
```

Comment:
Probably this was the reason why `threat-model` is required: if `self-assessment-created` is required, does this mean also `self-assessment` becomes required?

Reply:
Probably I just need to accept the logic will be always fallacious in a YAML schema: I wanted to avoid a scenario where a user could add `evidence-url` without adding `threat-model-created`. But we can do this "better": we can remove the boolean value, and leave only other fields (comments, evidence-url). I can work on this :)